docs: update dg use cases (#289)
AY2324S2-CS2103T-F13-1/tp: docs/DeveloperGuide.md

<span class="semi-bold">MSS</span>

1. User requests to add a student, providing the name and NUSNet ID as compulsory information, with the phone number, email, major and tags being optional.

   Use case ends.

</box>

<box no-icon type="success" light>

**Use case: Editing a Student**

<span class="semi-bold">MSS</span>

1. User requests to edit an existing student, indicating the student the user wish to edit. The user provides the update details for the student and remaining details are unchanged.

> 

Reviewer: Perhaps, this use case is a bit incomplete, we can add that TAPro successful edits the student details.

<box no-icon type="success" light>

**Use case: Find a Student by Name**

<span class="semi-bold">MSS</span>

1. User requests to find all students by one or more keywords in the student's name.

>     Use case ends.

Reviewer: This use case is a bit incomplete, we should state that TAPro shows the students that are found.

We can also add an extension that no result is found, so TAPro shows no students.

> 

Reviewer: Can we add week number here: `by providing the student's NUSNet ID and week number to mark/unmark the attendance for`.

And then TAPro marks/unmarks the attendance for that student for that week.

We should also add extensions if the student doesn't exist, or if the week doesn't exist, or if the attendance is already marked/unmarked.

Author: Very good point

<box no-icon type="success" light>

**Use case: Clear All Data**

<span class="semi-bold">MSS</span>

1. User requests to purge all data from TAPro.

>    Use case ends.

Reviewer: We can add that TAPro successfully clears all data.

> 

Reviewer: Perhaps we may want to add that TAPro successfully does something.

Author: I am not sure what you mean titus, do you mind elaborating?

Author: I see. I also moved the section on top of addstu use case.

<box no-icon type="success" light>

> **: Use case: Retrieve a previous successful command input**

Reviewer: ```suggestion
**Use case: Retrieve a previous successful command input**
```

<span class="semi-bold">MSS</span>

1. User focuses on the command box.

2. User presses the retrieve previous command hotkey (e.g. up arrow key).

> 3. The previous successful command input is shown in the command box.

Reviewer: ```suggestion
3. TAPro displays the successful command input in the command box.
```

   Use case ends.

<span class="semi-bold">Extensions</span>

* 3a. No previous successful command input is available.

    * 3a1. No action is taken.

    * Use case ends.

</box>

<box no-icon type="success" light>

> **: Use case: Retrieve a more recent successful command input**

Reviewer: ```suggestion
**Use case: Retrieve a more recent successful command input**
```

<span class="semi-bold">MSS</span>

1. User focuses on the command box.

2. User presses the retrieve next command hotkey (e.g. down arrow key).

> 3. The more recent successful command input is shown in the command box.

Reviewer: ```suggestion
3. TAPro displays a more recent successful command input in the command box.
```

   Use case ends.

<span class="semi-bold">Extensions</span>

* 3a. No successful command input that is more recent is available.

>     * 3a1. No action is taken.

Reviewer: Technically, the text in the command box is blank if no more recent command input is available.
